### PR TITLE
feat: clear balance cache after usage

### PR DESCRIPTION
**Pull request by alaibe**

### CODE REVIEW

**Comment by ghost** [comment]
### Pull Request Checklist
- [ ] Have you updated the documentation, if impacted (e.g. [docs.status.im](https://status.im/docs/))?
- [ ] Have you tested changes with mobile?
- [ ] Have you tested changes with desktop?

File: services/wallet/transfer/commands.go [hunk 1]
```diff
@@ -305,7 +305,7 @@ func (c *controlCommand) Run(parent context.Context) error {
 			BlockNumber: target,
 		})
 	}
-
+	bCache.Clear()
```
Comment (IvanBelyakoff) [inline services/wallet/transfer/commands.go]: I think it is safe to clear even earlier - at line 272 before loading transfers

File: services/wallet/transfer/balance_cache.go [hunk 2]
```diff
@@ -24,9 +30,19 @@ type balanceCache struct {
...
+func newBalanceCache() *balanceCache {
+	return &balanceCache{
+		balances:     make(map[common.Address]map[uint64]*big.Int),
+		nonces:       make(map[common.Address]map[uint64]*int64),
+		nonceRanges:  make(map[common.Address]map[int64]nonceRange),
+		sortedRanges: make(map[common.Address][]nonceRange),
+	}
+}
+func (b *balanceCache) Clear() {
+	b.balances = make(map[common.Address]map[uint64]*big.Int)
```
Comment (IvanBelyakoff) [inline services/wallet/transfer/balance_cache.go]: I think that we might want to keep first and last items for nonces and balances, as at least for OnDemand strategy, one side of the checked block range should hit the cache on next check.

**Review by IvanBelyakoff — APPROVED** [review]
LGTM

**Pull request merged**